### 1.Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16.  Найти доверительный интервал для оценки математического ожидания с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

$$T_{1,2} = {M} \pm \frac{sigma}{\sqrt{n}} \cdot{Z}$$

In [25]:
from math import sqrt

In [26]:
M = 80
sigma = 16
n = 256
z = 1.96

In [28]:
T1 = M + (sigma / sqrt(n)) * z
T1

81.96

In [29]:
T2 = M - (sigma / sqrt(n)) * z
T2

78.04

**Ответ:** [78.04, 81,96]

### 2 В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [30]:
list_x = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]

In [33]:
list_mean = sum(list_x) / len(list_x)
list_mean

6.589999999999999

In [35]:
n = len(list_x)
s = 0
for i in list_x:
    s += (i - list_mean)**2   
sigma = (s / n)**0.5
print(f'Среднее квадратичное отклонение = {sigma}')

Среднее квадратичное отклонение = 0.42766809560686203


In [36]:
se = sigma / sqrt(n)
se

0.13524052647043344

In [38]:
t_students = 2.2622

In [43]:
se * t_students

0.3059411189814145

In [40]:
T1 = list_mean + se * t_students
T1

6.895941118981414

In [42]:
T2 = list_mean - se * t_students
T2

6.284058881018584

**Ответ:** [6.28, 6.9] 

$$6.59 \pm 0.31$$

### 3 Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

1. Формулируем нулевую гипотезу: 17,5 = 17

2. Альтернативная гипотеза: 17,5 > 17

3. Выбираем стат критерий (считаем, что наше распределение нормальное, 5000 > n > 30, а дисперсию мы знаем заранее, значит выбираем Z-критерий)

$${sigma} = {sqrt(D)}$$

$${se} = \frac{sigma} {sqrt(n)}$$

$${Z} = (\overline{x} - {M}) / {se}$$

In [44]:
sigma = sqrt(4)
sigma

2.0

In [45]:
se = sigma / sqrt(100)
se

0.2

In [46]:
Z = (17.5 - 17) / se
Z

2.5

Табличное Z-значение  для a = 0.05 одностороннего = 2.575. Т.о. найденное Z(2,5) < a, и мы отвергаем нулевую гипотезу.

**Ответ:** выявлены статистические значимые различия средних значений диаметра шарика для подшипников (p < 0.05). Однако, учитывая близкие значения (2.5 и 2.575), рекомендую перепроверить данные, на выборке побольше.

### 4. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [47]:
cookies = [202, 203, 199, 195, 197, 201, 200, 204, 194, 190]

In [48]:
mean_cookies = sum(cookies) / len(cookies)
mean_cookies

198.5

1. Формулируем нулевую гипотезу: 198.5 = 200

2. Альтернативная гипотеза: 198.5 != 200

3. Выбираем стат критерий (считаем, что наше распределение нормальное,  n < 30, а дисперсию мы не знаем заранее, значит выбираем t-критерий)


a = 0.005 (1% разделенный на два)

табличное t-значение = 3.2498

In [59]:
t_students = 3.2498

In [56]:
n = len(cookies)
s = 0
for i in cookies:
    s += (i - mean_cookies)**2   
sigma = (s / (n - 1))**0.5
print(f'Среднее квадратичное отклонение = {sigma}')

Среднее квадратичное отклонение = 4.453463071962462


In [57]:
se = sigma / sqrt(n)
se

1.4083086782851737

In [60]:
t_students_n = (mean_cookies - 200) / se
t_students_n

-1.0651074037450896

Т.о t-наблюдаемое оказалось меньше t-табличного, значит мы принимаем нулевую гипотезу

**Ответ:** утверждение продавца верно.

In [62]:
from scipy.stats import *

y = cookies
x = [200, 200, 200, 200, 200, 200, 200, 200, 200, 200] 
 
ttest_ind(x, y)

Ttest_indResult(statistic=1.0651074037450894, pvalue=0.30090695525976713)